pkg/apis/pipeline/v1beta1/pipelinerun_types.go

```diff
@@ -153,17 +153,11 @@ func (pr *PipelineRun) GetNamespacedName() types.NamespacedName {
 }
 
 // IsTimedOut returns true if a pipelinerun has exceeded its spec.Timeout based on its status.Timeout
-func (pr *PipelineRun) IsTimedOut() bool {
-	return pr.HasTimedOut()
-}
-
-// HasTimedOut returns true if a pipelinerun has exceeded its spec.Timeout based on its status.Timeout
-func (pr *PipelineRun) HasTimedOut() bool {
```

Comment:
I was very confused why we had 2 functions with pretty much the same name doing what seemed like the same thing so i tracked it back to https://github.com/tektoncd/pipeline/commit/49ebeb6b63a1a957b1c6fa9078bda0c3d352d7c3 and it seems like @afrittoli  was introducing an interface that both PipelineRuns and Runs (custom tasks) would comply with - and im guessing that since Runs already had `HasTimedOut` he decided to standarize on that vs `IsTimedOut`?

Definitely makes sense to me to just use one or the other but I'm wondering if removing HasTimedOut breaks something that isnt covered by a test - it seems like this would make PipelineRuns no longer conform to the RunsToCompletion interface which might interfere with how events are emitted? (I would hope there would be a test covering this if so 🙏 )

Reply:
Thanks for looking into this! It looks like the RunsToCompletion interface was removed in a [later commit](https://github.com/tektoncd/pipeline/commit/51df0a8fd0f3c4ec05c67d4f75f4bb8f68946b41), so I think removing this function should be OK.

Reply:
we could move this refactor (removing the `HasTimedOut`) into its own commit, and focus this commit on using pr.Timeouts.Pipeline - what do you think?

Reply:
I think that probably isn't necessary, but I'm happy to do it if you'd prefer